cwms-data-api/src/test/java/cwms/cda/api/BasinControllerTestIT.java

```diff
@@ -1,59 +1,339 @@
+				.queryParam(FAIL_IF_EXISTS, "false")
+				.when()
+				.redirects().follow(true)
+				.redirects().max(3)
+				.post("basins/")
+				.then()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.assertThat()
+				.statusCode(is(HttpServletResponse.SC_CREATED))
+		;
+
+		String office = BASIN.getBasinId().getOfficeId();
+
+		// Retrieve basin and assert that it exists
+		given()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.accept(Formats.JSONV1)
+				.queryParam(Controllers.OFFICE, office)
+				.when()
+				.redirects().follow(true)
+				.redirects().max(3)
+				.get("basins/" + BASIN.getBasinId().getName())
+				.then()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.assertThat()
+				.statusCode(is(HttpServletResponse.SC_OK))
+				.body("basinId.name", equalTo(BASIN.getBasinId().getName()))
+				.body("basinId.officeId", equalTo(BASIN.getBasinId().getOfficeId()))
+				.body("sortOrder", equalTo(BASIN.getSortOrder().floatValue()))
+				.body("totalDrainageArea", equalTo(BASIN.getTotalDrainageArea().floatValue()))
+				.body("contributingDrainageArea", equalTo(BASIN.getContributingDrainageArea().floatValue()))
+				.body("parentBasinId.name", equalTo(BASIN.getParentBasinId().getName()))
+				.body("parentBasinId.officeId", equalTo(BASIN.getParentBasinId().getOfficeId()))
+				.body("areaUnit", equalTo(BASIN.getAreaUnit()))
+				.body("primaryStreamId.name", equalTo(BASIN.getPrimaryStreamId().getName()))
+				.body("primaryStreamId.officeId", equalTo(BASIN.getPrimaryStreamId().getOfficeId()))
+		;
+
+		// Delete basin
+		given()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.accept(Formats.JSONV1)
+				.queryParam(Controllers.OFFICE, office)
+				.queryParam("deleteAction", DELETE_ACTION)
+				.header(AUTH_HEADER, user.toHeaderValue())
+				.when()
+				.redirects().follow(true)
+				.redirects().max(3)
+				.delete("basins/" + BASIN.getBasinId().getName())
+				.then()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.assertThat()
+				.statusCode(is(HttpServletResponse.SC_NO_CONTENT))
+		;
+
+		// Retrieve basin and assert that it does not exist
+		given()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.accept(Formats.JSONV1)
+				.queryParam(Controllers.OFFICE, office)
+				.when()
+				.redirects().follow(true)
+				.redirects().max(3)
+				.get("basins/" + BASIN.getBasinId().getName())
+				.then()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.assertThat()
+				.statusCode(is(HttpServletResponse.SC_NOT_FOUND))
+		;
+	}
+
+	@Test
+	void test_update_does_not_exist() {
+		TestAccounts.KeyUser user = TestAccounts.KeyUser.SWT_NORMAL;
+		// Create basin
+		given()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.accept(Formats.JSONV1)
+				.contentType(Formats.JSONV1)
+				.queryParam(Controllers.OFFICE, user.getOperatingOffice())
+				.queryParam(Controllers.NAME, "newFakeName")
+				.header(AUTH_HEADER, user.toHeaderValue())
+				.when()
```

Comment:
the indentation should be
given
----setup
when
----setup
then
----setup
----assertThat
----assertions

where each - is a space

Reply:
Addressed in earlier commit